OmegaExamples/src/generative-models.jl

```diff
@@ -0,0 +1,412 @@
+md"The key idea is that we have primitive operations that describe not only deterministic functions (like `and`) but stochastic operations. For example, sampling from a Bernoulli distribution can be thought of as simulating a (possibly biased) coin toss:"
+
+# ╔═╡ 3c73f032-eedf-4d99-8489-a98ccf4b8805
+coin_toss = 1 ~ Bernoulli()
+
+# ╔═╡ 61f4d77b-eecf-4cca-ac01-944efe58d9bb
+randsample(coin_toss)
+
+# ╔═╡ 5a10bc0d-d676-4941-a1d5-41790ef255ec
+md"Run this program a few times. You will get back a different sample on each execution."
+
+# ╔═╡ 0ce18f90-f2ef-490a-a7c4-40a6ccfa214a
+# Describe what the function coin_toss does?
+
+# ╔═╡ 2ad90265-8ef2-480f-81b0-7ed1a674011b
+md"Each time you run a program you get a sample by simulating the computations and random choices that the program specifies. If you run the program many times, and collect the values in a histogram, you can see what a typical sample looks like:"
+
+# ╔═╡ 9504162b-52e1-476d-8218-0afc2555e278
+histogram(randsample(coin_toss, 1000), bins = 1)
+
+# ╔═╡ 15920581-2258-42e1-ba73-da82b2baa7e0
+md"As you can see, the result is an approximately uniform distribution over true(1) and false(0).
+This way we can construct more complex expressions that describe more complicated sampling processes. For instance, here we describe a process that samples a number adding up several independent Bernoulli distributions:"
+
+# ╔═╡ 3f04b226-a38a-4804-8691-59332bfc8728
+b_sum(ω) = (1 ~ Bernoulli())(ω) + (2 ~ Bernoulli())(ω) + (3 ~ Bernoulli())(ω)
+
+# ╔═╡ db8d667c-5ad7-4deb-b020-13d645bedbf9
+randsample(b_sum)
+
+# ╔═╡ 5444fd8d-8ff3-4b30-9942-067dca8a1126
+md"We have constructed a stochastic function that adds three random numbers each time it is called so that we may sample multiple times from it. We can construct such complex stochastic functions from the primitive ones."
+
+# ╔═╡ 09595cc9-b08a-4465-a3e0-5d8e280530a2
+histogram(randsample(b_sum, 1000), bins = 3)
+
+# ╔═╡ 6aa46f74-b3e2-4d12-abd0-90759023ecc5
+md"Complex functions can also have other arguments. Here is a stochastic function that will only sometimes double its input:"
+
+# ╔═╡ 0974a6a9-859f-44a9-a17a-5ef90c789ce5
+begin
+	noisyDouble(ω, x) = (1 ~ Bernoulli())(ω) ? 2*x : x
+	noisyDouble(x) = ω -> noisyDouble(ω, x)
+end
+
+# ╔═╡ 11c2d2b5-0c14-4b23-9282-083ac6f07f6f
+randsample(noisyDouble(3))
+
+# ╔═╡ c4956dd3-f4fc-4112-b857-423f19c3209e
+# Explain omega? Multiple dispatch?
+
+# ╔═╡ c697307c-9d49-43a7-a379-f383780a1248
+md"By using higher-order functions we can construct and manipulate complex sampling processes. We use the ternary operator `condition ? if-true : if-false` to induce hierarchy. A good example comes from coin flipping…"
+
+# ╔═╡ 277b383e-c909-4424-91f1-f0f00b3531b4
+md"## Example: Flipping Coins"
+
+# ╔═╡ f58cd6ff-9023-4dc7-9213-2279f5aebf72
+md"The following program defines a fair coin, and flips it 20 times:"
+
+# ╔═╡ 3c5e0f1d-0379-4c92-817d-4e6ef2d7ed8e
+fairCoin(ω) = (1 ~ Bernoulli())(ω) ? 'h' : 't'
```

Comment:
By convention, we don't use camelCase in Julia (except for types).  Use `fair_coin` or `faircoin` instead